subiquity/common/filesystem/tests/test_sizes.py

```diff
@@ -74,3 +77,70 @@ def test_efi(self):
         bootfs_size = get_bootfs_size(disk_size)
         self.assertTrue(bootfs_scale.maximum > bootfs_size)
         self.assertTrue(bootfs_size > bootfs_scale.minimum)
+
+
+class TestCalculateGuidedResize(unittest.TestCase):
+    def test_ignore_nonresizable(self):
+        actual = calculate_guided_resize(
+                part_min=-1, part_size=100 << 30, install_min=10 << 30)
+        self.assertIsNone(actual)
+
+    def test_too_small(self):
+        actual = calculate_guided_resize(
+                part_min=95 << 30, part_size=100 << 30, install_min=10 << 30)
+        self.assertIsNone(actual)
+
+    def test_even_split(self):
+        # 8 GiB * 1.25 == 10 GiB
+        size = 10 << 30
+        actual = calculate_guided_resize(
+                part_min=8 << 30, part_size=100 << 30, install_min=size)
+        self.assertEqual(size, actual.minimum)
```

Comment:
Did you think about writing these as a comparison of GuidedResizeValues instances?

Reply:
Done